### ISSUE TITLE
Random results when save model in Python 3.5 and load in Python 2.7

### ISSUE THREAD

**Issue by Zimba96** [keras-team/keras#9925]
Hi there,

I'm using Keras 2.1.5 with Tensorflow-gpu 1.7.0 as backend on ubuntu 16.04 (Kernel 4.4).
I've got a model trained in Python 3.5 saved with model.save("path"). Evaluating this model on Python 3.5 results in an accuracy of 99%. Futhermore, loading the model with Python 3.5 on another machine works perfectly fine.

But I've got to load it on Python 2.7. Doing so (model=load_model("path") -> model.evaluate(...)), the evaluation results in 33% accuracy.

I've tried to save it in json and YAML as well as building the architecture and loading the weights on it...nothing works.

I haven't found any thread with this problem so far...

Hope that someone has an idea of what's going on!

Thanks in advance!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by benni-r** [comment]
I have the same issue. My solution is training the model in Python 2.7 as well, but I'm not happy with that.
Is there a possibility in saving the model in another file-format or saving the tensorflow-checkpoints separately?
